xrpl4j-core/src/main/java/org/xrpl/xrpl4j/codec/binary/types/AmountType.java

```diff
@@ -166,22 +178,45 @@ public AmountType fromJson(JsonNode value) throws JsonProcessingException {
         rawBytes[0] |= 0x40;
       }
       return new AmountType(UnsignedByteArray.of(rawBytes));
-    }
+    } else if (!value.has("mpt_issuance_id")) {
+      // IOU Amount
+      Amount amount = objectMapper.treeToValue(value, Amount.class);
+      BigDecimal number = new BigDecimal(amount.value());
 
-    Amount amount = objectMapper.treeToValue(value, Amount.class);
-    BigDecimal number = new BigDecimal(amount.value());
+      UnsignedByteArray result = number.unscaledValue().equals(BigInteger.ZERO) ?
+        UnsignedByteArray.fromHex(ZERO_CURRENCY_AMOUNT_HEX) :
+        getAmountBytes(number);
 
-    UnsignedByteArray result = number.unscaledValue().equals(BigInteger.ZERO) ?
-      UnsignedByteArray.fromHex(ZERO_CURRENCY_AMOUNT_HEX) :
-      getAmountBytes(number);
+      UnsignedByteArray currency = new CurrencyType().fromJson(value.get("currency")).value();
+      UnsignedByteArray issuer = new AccountIdType().fromJson(value.get("issuer")).value();
+
+      result.append(currency);
+      result.append(issuer);
+
+      return new AmountType(result);
+    } else {
+      // MPT Amount
+      MptAmount amount = objectMapper.treeToValue(value, MptAmount.class);
+
+      if (FluentCompareTo.is(amount.unsignedLongValue()).greaterThan(UnsignedLong.valueOf(Long.MAX_VALUE))) {
+        throw new IllegalArgumentException("Invalid MPT amount: given value requires 64 bits, only 63 allowed.");
```

Comment:
The current error wording was slightly confusing to me. What do you think of this instead?

```suggestion
        throw new IllegalArgumentException("Invalid MPT amount. Maximum MPT value is (2^63 - 1)");
```

Reply:
changed in https://github.com/XRPLF/xrpl4j/pull/556/commits/73e282ed78fa791fd33d801ca76174b65eae42bf